### PR TITLE
feat: adding Cypress Component Tests

### PR DESCRIPTION
**Pull request by JessicaSachs**
This branch is a draft. It attempts to setup Cypress Component Tests with the Real World App.

### CODE REVIEW

**Comment by dmtrKovalenko** [comment]
I fixed this example. Making a PR in component testing to include `FAST_REFRESH=false` inside react-scripts plugin. 

> <AlertBar /> looks like not the best component that should be covered using component tests, but it works

File: package.json [hunk 1]
```diff
@@ -17,6 +17,8 @@
   "dependencies": {
     "@auth0/auth0-react": "1.3.0",
     "@aws-amplify/ui-react": "0.2.38",
+    "@cypress/react": "^5.1.0-alpha.1",
+    "@cypress/webpack-dev-server": "^1.0.4",
```
Comment (kevinold) [inline package.json:21]: @JessicaSachs @dmtrKovalenko Should these be moved to devDependencies?

Comment (dmtrKovalenko) [reply package.json:21]: This doesn't really matter for apps right? Or you are using yarn install --production for server? 

But I don't care we can move it :)

Comment (kevinold) [reply package.json:21]: @dmtrKovalenko It does for vite tooling, which we will be merging soon as an option during development (https://github.com/cypress-io/cypress-realworld-app/pull/741).  Please move these to `devDependencies`.

Comment (JessicaSachs) [reply package.json:21]: For Vite tooling, we'll be swapping this to the `@cypress/vite-dev-server` package. Trying out the webpack version first 😻 . We have an esm build as well, but yes, these should be devDeps.

**Comment by JessicaSachs** [comment]
@dmtrKovalenko I updated this PR with the new versions and tried to implement the first test (AlertBar.tsx) -- I run into the ReactDOM Issue (Jira: [CT-378](https://cypress-io.atlassian.net/browse/CT-378)).

To get it to fail you can do `yarn cypress:open:ct` and try running the AlertBar.test.tsx spec

**Comment by dmtrKovalenko** [comment]
So I did implement one test in my commit for this example. Should I pick this up?

**Comment by JessicaSachs** [comment]
@dmtrKovalenko That would be awesome. 👍

File: src/components/__tests__/AlertBar.test.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import ReactDom from "react-dom";
+import { mount } from "@cypress/react";
+import { snackbarMachine } from "../../machines/snackbarMachine";
+import { useMachine } from "@xstate/react";
+
+import AlertBar from "../AlertBar";
+
+describe("smoke", () => {
+  it("works", () => {
+    const [, , snackbarService] = useMachine(snackbarMachine);
```
Comment (dmtrKovalenko) [inline src/components/__tests__/AlertBar.test.tsx:13]: BTW @JessicaSachs that is not the ReactDom issue. You can not just call the react hook inside the test

**Comment by kevinold** [comment]
@JessicaSachs @dmtrKovalenko What's the status of this branch?

**Pull request closed without merging**